source/objects/RenderTarget.cpp

```diff
@@ -209,6 +216,197 @@ Pixel RenderTarget::SampleBilinear(float u, float v, uint8_t mip) const
 	};
 }
 
+static bool verifyInRoot(const fs::path& target, const fs::path& root) {
+	// We can assure that the user did not go out of bounds by checking that the root is equal to the working directory.
+	std::string finalPathString = target.string();
+	std::string workingDirString = root.string();
+
+	std::string rootSubStr = finalPathString.substr(0, workingDirString.length());
+	if (rootSubStr != workingDirString) {
+		return false; // User exited the root directory.
+	}
+
+	return true;
+}
+
+bool RenderTarget::Load(const char* filepath, bool generateMips) __attribute__((optnone))
+{
+	if (!IsValid()) return false;
+
+	int imageWidth = 0;
+	int imageHeight = 0;
+	int channels = CHANNELS[static_cast<uint8_t>(mFormat)];
+
+	uint8_t* rtData = GetRawData();
+
+	// Sandbox the filepath to data/vistrace
+	fs::path workingDir(fs::current_path());
+	workingDir += "/garrysmod/data/vistrace";
+	workingDir = fs::absolute(workingDir);
+	workingDir = workingDir.make_preferred();
+
+	if (!fs::is_directory(workingDir)) {
+		return false;
+	}
+	
+	fs::path filename(filepath);
+	filename = filename.make_preferred();
```

Comment:
Dont need to make preferred this (doesnt even look like you need this variable in the first place)
Make preferred finalPath instead